sisl/_namedindex.py

```diff
@@ -106,6 +106,20 @@ def __contains__(self, name):
         """ Check whether a name exists in this group a named group """
         return name in self._name
 
+    def sub(self, index):
```

Comment:
I should revise what NamedIndex does in these cases.

Currently the `remove` function works in-place. But your sub does not.
I'll have to think about this. Also, I think that a named-index should be allowed, even if it has no entries.

Reply:
In-place or not, whichever you like -- it doesn't really matter I think. I don't think there's a usecase for sub outside of when it's used in geometry.

I personally don't like 'leftovers'. I am similarly annoyed that there can be (and often is) an Atom in the Atoms which is not present in the geometry. But 'empty' names could be left behind for consistency with this behaviour. (I might send a PR with a `.clean()` or similar function later then)

(regarding consistency: Atoms.sub is not in-place either, I believe?)

Reply:
The leftover in `Atoms` has its use cases (primarily targetted Siesta). And in principle it could be nice to have assignments of names that "disappear" and then you at least know that you *had* some name associated, but have (un)-intentionally removed it.

`Atoms.reduce` removes un-used atoms, so it is there.

Regarding consistency, `Atoms.sub` is not in-place, just as I wrote above. `sub`/`remove` should create new instances while `delete` should do it *in-place*.